File: Python3.x/406-Queue_Reconstruction_by_Height.py
# Runtime: 76 ms, faster than 99.32% of Python3 online submissions for Queue Reconstruction by Height.
# Memory Usage: 12.9 MB, less than 17.14% of Python3 online submissions for Queue Reconstruction by Height.
class Solution:
    def reconstructQueue(self, people: 'List[List[int]]') -> 'List[List[int]]':
        people = sorted(people, key = lambda x: (-x[0], x[1]))  # 高度递减，k增加 排序
        result = []
        for item in people:
             result.insert(item[1], item)  # 根据k前面不低于自己身高人数 插入
        return result

File: Algorithms/C++/406-Queue_Reconstruction_by_Height.cpp
// Runtime: 136 ms, faster than 13.79% of C++ online submissions for Queue Reconstruction by Height.
// Memory Usage: 25.6 MB, less than 11.60% of C++ online submissions for Queue Reconstruction by Height.
class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        sort(people.begin(), people.end(), [](vector<int> p1, vector<int> p2){
            return p1[0] > p2[0] || (p1[0] == p2[0] && p1[1] < p2[1]);
        });
        vector<vector<int>> res;
        for (vector<int> person : people)
            res.insert(res.begin() + person[1], person);
        return res;
    }
};
